### PR TITLE
Added Session 3

### LINKED ISSUE #7: Create Session Three

### PR DESCRIPTION
**Pull request by pavly-gerges**
## Contents : 
- Simple Arrays and simple listing.
- Iteration over items using loops.
- Reverse iteration using progressions.
- Multidimensional Arrays.
- Hybrid Lists.
- Using 'is' to check if an item is of a specific type.
- ArrayLists.
## <s>WIP : </s>
<s>- ArrayLists.</s>

### CODE REVIEW

**Comment by pavly-gerges** [comment]
Session 3 is done, please revise it !!

**Review by AhmedShehatah — COMMENTED** [review]
what you wrote are lists we need to add arrays at first so we can explan fixed size of arrays 
```.kt
val arr = Array<String>(10)
```
and what is the meaning of default values 
```.kt
val arr = Array<String>(10){"default value"}
```

**Comment by pavly-gerges** [comment]
> what you wrote are lists we need to add arrays at first so we can explan fixed size of arrays
> 
> ```kotlin
> val arr = Array<String>(10)
> ```
> 
> and what is the meaning of default values
> 
> ```kotlin
> val arr = Array<String>(10){"default value"}
> ```

I will add the `Arrays` before lists, using both the `Array` constructor and the initialize function and `arrayOf()` function.

**Comment by pavly-gerges** [comment]
> what you wrote are lists we need to add arrays at first so we can explan fixed size of arrays
> 
> ```kotlin
> val arr = Array<String>(10)
> ```

This is not a valid statement, you have to use an init lambda function.

File: src/kotlinSessions/SessionThree.kt [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    // to define an Array use the type Array<Type> which will define an array of a specific type
+    val studentNames: Array<String> = arrayOf("Maged", "Abdo")
+    studentNames.forEach {
+        println(it)
+    }
+    // another way to define arrays of fixed size.
+    // the block {} is called an initialize phase for the constructor, it creates default values for our array while initializing it using the indices
+    val students: Array<String> = Array(3) {
+        index: Int -> "Name"
+    }
```
Comment (AhmedShehatah) [inline src/kotlinSessions/SessionThree.kt:23]: you can just type
```.kt
val students: Array<String> = Array(3){"name"}
```

File: src/kotlinSessions/SessionThree.kt [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    students.forEach {
+        println(it)
+    }
+    // print our size
+    println("Array Size = ${students.size}")
+
+    //printing our items using ranges
+    for (index in students.indices step 1) {
+        println(students[index])
+    }
```
Comment (AhmedShehatah) [inline src/kotlinSessions/SessionThree.kt:40]: no need for step 1 because it is the init value

File: src/kotlinSessions/SessionThree.kt (hunk 1 above)
Comment (pavly-gerges) [reply src/kotlinSessions/SessionThree.kt:23]: I know, but you should be able to show them the full version.

File: src/kotlinSessions/SessionThree.kt (hunk 2 above)
Comment (pavly-gerges) [reply src/kotlinSessions/SessionThree.kt:40]: I just want to remind them that we can control the loop steps using the step function.

File: src/kotlinSessions/SessionThree.kt (hunk 1 above)
Comment (pavly-gerges) [reply src/kotlinSessions/SessionThree.kt:23]: The discussion is : we want to assign each item by their indexes to this return value.

**Pull request merged**